action.yml

```
@@ -113,7 +115,7 @@ runs:
       shell: bash
 
     - uses: actions/github-script@v6
-      if: ${{ inputs.post_comment == 'true' || ( inputs.post_comment == 'nonzero' && steps.check.outputs.returncode != 0 ) }}
+      if: ${{ inputs.post_comment == 'true' || inputs.update_comment == 'true' || ( inputs.post_comment == 'nonzero' && steps.check.outputs.returncode != 0 ) }}
```

```suggestion
      if: ${{ inputs.post_comment == 'true' || inputs.post_comment == 'update' || ( inputs.post_comment == 'nonzero' && steps.check.outputs.returncode != 0 ) }}
```

How about including the `update` option inside `post_comment` instead of adding a new input?
Having both `post_comment` and `update_comment` kind of makes me think that both has to be set to `true`

---

Yes, this seem like a better way to handle it. I'll push a change.

---

4258c25b57ee60f26ffca48f3749e5a3f4ad5f21 - currently you could not combine `nonzero` and `update` but I don't think would make sense to do so, so that should be fine

---

Yeah, using both `nonzero` and `update` mean the comment won't get updated if the check passes